main.go

```diff
@@ -32,36 +42,42 @@ func main() {
 		id.ClientVersion += "+relay"
 	}
 
-	if *pprofport > 0 {
-		fmt.Printf("Running metrics server on port: %d\n", *pprofport)
-		go metrics.SetupMetrics(*pprofport)
+	if *metricsPort > 0 {
+		fmt.Printf("Running metrics server on port: %d\n", *metricsPort)
+		go metrics.SetupMetrics(*metricsPort)
 	}
 
 	if *inmem {
 		*dbpath = ""
 	}
 
-	runOpts := runner.Options{
+	// Allow short keys. Otherwise, we'll refuse connections from the bootsrappers and break the network.
+	// TODO: Remove this when we shut those bootstrappers down.
+	crypto.MinRsaKeyBits = 1024
+
+	opts := hydra.Options{
 		DatastorePath: *dbpath,
 		Relay:         *relay,
 		BucketSize:    *bucketSize,
+		GetPort:       utils.PortSelector(*portBegin),
+		NSybils:       *many,
+		BsCon:         *bootstrapConcurency,
+		Stagger:       *stagger,
+	}
+
+	hy, err := hydra.NewHydra(opts)
+	if err != nil {
+		log.Fatalln(err)
 	}
 
-	if *many == -1 {
-		err := runner.RunSingle(runOpts)
-		if err != nil {
-			log.Fatalln(err)
-		}
+	go httpapi.ListenAndServe(hy, httpAPIAddr)
 
-		return
+	var peers []peer.ID
+	for _, nd := range hy.Sybils {
+		peers = append(peers, nd.Host.ID())
 	}
 
-	runOpts.GetPort = utils.PortSelector(*portBegin)
-	runOpts.NSybils = *many
-	runOpts.BsCon = *bootstrapConcurency
-	runOpts.Stagger = *stagger
-	err := runner.RunMany(runOpts)
-	if err != nil {
-		log.Fatalln(err)
+	if !*noUI {
+		go ui.NewUI(peers, uiopts.Start(start), uiopts.MetricsPort(*metricsPort))
```

Comment:
Why do this filtering here and not inside the UI?

Reply:
hmm...I guess I don't see the need to instantiate a thing if it's not going to do anything?